fix(Provider): staticStyles should be executed with the merged siteVariables

mnajdova opened the pull request:
This PR aims to fix https://github.com/stardust-ui/react/issues/526 
# Note
Important thing to note is that, per one Provider we will execute the staticStyles only once, when they are the first time applied.

Commit:
wip

Commit:
-added logic for rendering the staticStyles if they are provided, just once in one Provider instance.

Commit:
Merge branch 'master' into fix/static-styles-site-var-param

layershifter commented on src/components/Provider/Provider.tsx:
```diff
@@ -23,7 +23,14 @@ export interface ProviderProps {
+    super(props)
+    this.staticStylesRendered = false
+  }
+
```
```suggestion
staticStylesRendered: boolean = false
```

You can set a value without `constructor`, or I am missing something?

mnajdova replied:
Yep, I will remove the constructor. I was trying different things, it is just leftover.

Commit:
-updated changelog

Commit:
-added for ensuring that staticStyles are executed just once in the Provider
-added test for ensuring that staticStyles are invoked with the merged siteVariables

kuzhelov-ms committed:
Merge branch 'master' into fix/static-styles-site-var-param

kuzhelov-ms committed:
adjust changelog entry
